packages/browser/src/tracing/browserTracingIntegration.ts

```diff
@@ -196,18 +211,17 @@ export const browserTracingIntegration = ((_options: Partial<BrowserTracingOptio
 
   /** Create routing idle transaction. */
   function _createRouteSpan(client: Client, startSpanOptions: StartSpanOptions): Span {
-    const { beforeStartSpan, idleTimeout, finalTimeout, childSpanTimeout } = options;
-
     const isPageloadTransaction = startSpanOptions.op === 'pageload';
 
     const finalStartSpanOptions: StartSpanOptions = beforeStartSpan
       ? beforeStartSpan(startSpanOptions)
       : startSpanOptions;
 
-    // If `beforeStartSpan` set a custom name, record that fact
     const attributes = finalStartSpanOptions.attributes || {};
 
-    if (finalStartSpanOptions.name !== finalStartSpanOptions.name) {
+    // If `finalStartSpanOptions.name` is different than `startSpanOptions.name`
+    // it is because `beforeStartSpan` set a custom name. Therefore we set the source to 'custom'.
+    if (startSpanOptions.name !== finalStartSpanOptions.name) {
```

Comment:
This is the change. I'm not sure how the tests were passing 🤔

Reply:
OMG 🤯

Reply:
I guess we never tested starting a pageload/navigation txn with `beforeStartSpan` 🤔